### PR TITLE
fix(SimplePagination) : Add total items count

### PR DESCRIPTION
**Pull request by herleraja**
Closes #

**Summary**

- add total item counts to the simple pagination component and hide custom range links if not required.

**Change List (commits, features, bugs, etc)**

- fix(SimplePagination) : Add total items count
- fix(DateTimePicker): hide custom range link if not required
- fix(List): return the parent id

**Acceptance Test (how to verify the PR)**

- DateTimePicker and SimplePagination component works without any issues

### CODE REVIEW

File: packages/react/src/components/SimplePagination/SimplePagination.jsx [hunk 1]
```diff
@@ -48,56 +60,63 @@ const SimplePagination = ({
...
+          {pageText ? `${pageText} ${page}` : pageOfPagesText(page, maxPage)}
+        </span>
+        {maxPage > 1 ? (
+          <>
+            <div
               className={
                 hasPrev
-                  ? `${iotPrefix}-simple-pagination-caret`
-                  : `${iotPrefix}-simple-pagination-caret-disabled`
+                  ? `bx--pagination__button bx--pagination__button--backward ${iotPrefix}-addons-simple-pagination-button`
```
Comment (sstone2423) [inline packages/react/src/components/SimplePagination/SimplePagination.jsx:78]: Use the included `prefix` for carbon's `bx` and the `addons` string isn't needed as the `iotPrefix` handles our unique prefixing. We should also try to follow BEM by adding `--` between the prefix and component name, and adding `__` between the main component and subcomponent names
```suggestion
                  ? `${prefix}--pagination__button bx--pagination__button--backward ${iotPrefix}--simple-pagination__button`
```

File: packages/react/src/components/SimplePagination/SimplePagination.jsx [hunk 2]
```diff
@@ -48,56 +60,63 @@ const SimplePagination = ({
...
+                  : `bx--pagination__button bx--pagination__button--backward ${iotPrefix}-addons-simple-pagination-button-disabled`
```
Comment (sstone2423) [inline packages/react/src/components/SimplePagination/SimplePagination.jsx:79]: We should also try to follow BEM by adding `--` between the prefix and component name, and adding `__` between the main component and subcomponent names. For specific behaviors or differences in the subcomponent, add `--`
```suggestion
                  `${prefix}---pagination__button ${prefix}--pagination__button--backward ${iotPrefix}--simple-pagination__button--disabled`
```

Comment (sstone2423) [reply packages/react/src/components/SimplePagination/SimplePagination.jsx:79]: It looks like the component was previously written like this , but it wouldn't hurt to update to the proper standard

File: packages/react/src/components/SimplePagination/SimplePagination.story.jsx [hunk 3]
```diff
@@ -25,6 +25,8 @@ export const Default = () => (
       page={number('page', 1)}
       maxPage={number('maxPage', 4)}
       onPage={action('onPage')}
+      totalItems={number('totalItems', 10)}
+      totalItemsText={text('total tems display text (totalItemsText)', 'Assets')}
```
Comment (sstone2423) [inline packages/react/src/components/SimplePagination/SimplePagination.story.jsx:29]: Small typo on `items`

**Review by sstone2423 — CHANGES_REQUESTED** [review]

**Review by sstone2423 — APPROVED** [review]

**Review by scottdickerson — APPROVED** [review]
LGTM

File: packages/react/src/components/SimplePagination/SimplePagination.jsx [hunk 4]
```diff
@@ -48,56 +60,63 @@ const SimplePagination = ({
...
-            }
-            role="button"
-            tabIndex={hasPrev ? 0 : -1}
-            onClick={hasPrev ? handlePrev : undefined}
-            onKeyDown={hasPrev ? (evt) => handleEnterKeyDown(evt, handlePrev) : undefined}
-          >
-            <CaretLeft16
-              dir="ltr"
-              description={prevPageText}
+    <div className={`${iotPrefix}--simple-pagination-container`} data-testid={testID}>
```
Comment (kevinsperrine) [inline packages/react/src/components/SimplePagination/SimplePagination.jsx:63]: Should these class name changes happen in a minor version? It could potentially create a lot of issues for people using xpath tests.

File: packages/react/src/components/List/List.jsx [hunk 5]
```diff
@@ -184,7 +184,7 @@ const List = forwardRef((props, ref) => {
           editingStyle={editingStyle}
           secondaryValue={secondaryValue}
           rowActions={rowActions}
-          onSelect={editingStyle ? () => {} : () => handleSelect(item.id)}
+          onSelect={editingStyle ? () => {} : () => handleSelect(item.id, parentId)}
```
Comment (davidicus) [inline packages/react/src/components/List/List.jsx:187]: This seems unrelated to the rest of these changes. Was this intentional?

File: packages/react/src/components/SimplePagination/SimplePagination.jsx [hunk 6]
```diff
@@ -48,56 +60,63 @@ const SimplePagination = ({
...
+          {pageText ? `${pageText} ${page}` : pageOfPagesText(page, maxPage)}
+        </span>
+        {maxPage > 1 ? (
+          <>
+            <div
               className={
                 hasPrev
-                  ? `${iotPrefix}-simple-pagination-caret`
-                  : `${iotPrefix}-simple-pagination-caret-disabled`
+                  ? `bx--pagination__button bx--pagination__button--backward ${iotPrefix}-addons-simple-pagination__button`
```
Comment (davidicus) [inline packages/react/src/components/SimplePagination/SimplePagination.jsx:78]: Is there a reason we are changing these styles. I agree that we shouldn't these. Let's keep these in and open an issue to remove these in V3 and add the version 3 tag

File: packages/react/src/components/SimplePagination/_simple-pagination.scss [hunk 7]
```diff
@@ -1,29 +1,34 @@
 @import '~carbon-components/scss/globals/scss/vars';
 @import '../../globals/vars';
 
-.#{$iot-prefix}-simple-pagination-container {
+.#{$iot-prefix}--simple-pagination-container {
   display: flex;
   height: $spacing-09;
-  justify-content: flex-end;
+  //justify-content: space-between;
```
Comment (davidicus) [inline packages/react/src/components/SimplePagination/_simple-pagination.scss:7]: Can we lose this commented out code?

**Review by davidicus — CHANGES_REQUESTED** [review]
Just a few comments

File: packages/react/src/components/List/List.jsx (hunk 5 above)
Comment (herleraja) [reply packages/react/src/components/List/List.jsx:187]: This was a fix for select function. In some place we are sending parentid back and some place we aren't.  So it's best to send this information so that we can use it in a callback function.

File: packages/react/src/components/SimplePagination/SimplePagination.jsx (hunk 6 above)
Comment (herleraja) [reply packages/react/src/components/SimplePagination/SimplePagination.jsx:78]: Sure. Let me revert these changes.
@sstone2423 FYI. Let's not edit styles property till V3

**Review by davidicus — APPROVED** [review]
